activerecord/lib/active_record/future_result.rb

```diff
@@ -77,12 +101,17 @@ def canceled?
       end
 
       def execute_or_wait
-        return unless pending?
-
-        @mutex.synchronize do
-          if pending?
-            execute_query(@pool.connection)
+        if pending?
+          start = Process.clock_gettime(Process::CLOCK_MONOTONIC, :millisecond)
```

Comment:
Do we need to handle JRuby here? IIRC Jruby doesn't support `Process.clock_gettime`. ActiveSupport has some examples of how we've dealt with this in the past. If none of the async functionality works for JRuby then we can leave this as is.

Reply:
Ah thanks for the pointer. I'll have a look at how active support does it. I don't see any reason why async queries wouldn't work on JRuby.

Reply:
Done.

Reply:
So for the record, apparently `Process.clock_gettime` is implemented by JRuby: https://github.com/ruby-concurrency/concurrent-ruby/pull/915